src/CMakeLists.txt

```diff
@@ -168,6 +168,34 @@ add_subdirectory(probabilistic)
 
 add_subdirectory(fuzzers)
 
+########################################################################
+## Build in the discovered external plugins
+
+foreach (plugin_dir ${BUILTIN_PLUGIN_LIST})
+    get_filename_component(plugin_name ${plugin_dir} NAME)
+
+    if(IS_DIRECTORY "${plugin_dir}/cmake")
+        set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${plugin_dir}/cmake")
+    endif()
+
+    # Setup the include path for built source artifacts.
+    include_directories(AFTER 
+                        ${CMAKE_CURRENT_BINARY_DIR}/plugins/${plugin_name}
+                        ${plugin_dir}/src)
```

Comment:
Agreed.  I didn't like that I essentially ended up encoding paths that may be conventions rather than requirements.  This may be where @ckreibich can come in since he's trying to clean up plugin definitions at the moment too.

@ckreibich do you have any thoughts about enforcing plugin layout more strictly than we do now?  I know that you're mostly focused on the scripts, but the cmake directory is another place with similar potential problems.  I think that we'd have to have a standardized location for the cmake directory though since typically looking for libraries happens prior to most of the plugin definition.

(also, the src/ directory shouldn't matter since that can be encoded by users wherever they want since it's encoding in the CMakeLists.txt file)

Reply:
The more I think about it, the more I think that the cmake/ directory is already encoded in the build system because when you build plugins, if you put cmake scripts into that directory they will automatically be found.

Reply:
> (also, the src/ directory shouldn't matter since that can be encoded by users wherever they want since it's encoding in the CMakeLists.txt file)

Yeah, I'm thinking that also -- a plugin's own `CMakeLists.txt` should have set up whatever include paths they need for their source-file locations.  So the question becomes: why does this logic need to alter `include_directories()` ?  The `${plugin_dir}/src` is a hard-coded path in our own `CMakeLists.txt` making an assumption about the the directory layout of the external plugin.

> The more I think about it, the more I think that the cmake/ directory is already encoded in the build system because when you build plugins, if you put cmake scripts into that directory they will automatically be found.

A guess is that it works in dynamic plugins because of this: https://github.com/zeek/cmake/blob/9132016bfb9ae2953830afaf65a8ef2057aa7a76/ZeekPluginDynamic.cmake#L163

But that won't apply when building them statically like this, which is why altering the `CMAKE_MODULE_PATH` here for parity like you had seems right, just still have the minor observation that Plugin Directory Layout docs don't mention this standardization.

Reply:
> The ${plugin_dir}/src is a hard-coded path in our own CMakeLists.txt making an assumption about the directory layout of the external plugin.

I think I may have over zealously included that path.  The one that I especially needed was the one from the prior line for the source artifacts from bif/binpac runs.  When I work on this again I'll double check if that include is necessary.